### PR TITLE
Spread even if nodes are not available

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Several changes to make spread scheduling work better under load:
- When nodes are not available, spread among feasible nodes.
- If grant_or_reject is true, don't spill back if the selected node is not available.
- Don't spill due to waiting for dependencies for spread tasks.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/raylet/scheduling/cluster_resource_scheduler.cc [hunk 1]
```diff
@@ -211,21 +213,34 @@ scheduling::NodeID ClusterResourceScheduler::GetBestSchedulableNode(
...
+                             requires_object_store_memory,
+                             task_spec.IsActorCreationTask(),
+                             exclude_local_node,
+                             &_unused,
+                             is_infeasible);
+
+  // If there is no other available nodes, prefer waiting on the local node
+  // since the local node is chosen for a reason (e.g. spread).
+  if (prioritize_local_node && !best_node.IsNil() &&
+      !IsSchedulableOnNode(best_node,
```
Comment (stephanie-wang) [inline src/ray/raylet/scheduling/cluster_resource_scheduler.cc:236]: What's the difference between this and the IsAvailable check used elsewhere?

**Review by stephanie-wang — CHANGES_REQUESTED** [review]
Had some questions about the fix. Also, can we add tests?

File: src/ray/raylet/scheduling/cluster_resource_scheduler.cc (hunk 1 above)
Comment (jjyao) [reply src/ray/raylet/scheduling/cluster_resource_scheduler.cc:236]: They are actually the same: different functions but same implementation. We should clean them up in a separate PR.

**Review by stephanie-wang — APPROVED** [review]
Nice!

**Pull request merged**